src/gui/http.go

```diff
@@ -74,33 +95,32 @@ func LaunchWebInterfaceHTTPS(host, staticDir string, daemon *daemon.Daemon, cert
 		return err
 	}
 
-	listener, err = tls.Listen("tcp", host, &tls.Config{Certificates: certs})
+	s.listener, err = tls.Listen("tcp", host, &tls.Config{Certificates: certs})
 	if err != nil {
 		return err
 	}
 
 	// Runs http.Serve() in a goroutine
-	serve(listener, NewGUIMux(appLoc, daemon), quit)
 	return nil
+
 }
 
-func serve(listener net.Listener, mux *http.ServeMux, q chan struct{}) {
-	go func() {
-		for {
-			if err := http.Serve(listener, mux); err != nil {
-				select {
-				case <-q:
-					return
-				default:
-				}
-				continue
+func (s *Server) Serve() {
+	mux := NewGUIMux(s.appLoc, s.daemon)
```

Comment:
Add logging here about http server starting and defer a "server closed" log statement, similar to other `Run` methods